src/OrchardCore.Modules/OrchardCore.Autoroute/Services/AutorouteAliasProvider.cs

```diff
@@ -26,14 +30,18 @@ public Task<string> GetContentItemIdAsync(string alias)
                     alias = "/" + alias;
                 }
 
-                string contentItemId;
-                if (_autorouteEntries.TryGetContentItemId(alias, out contentItemId))
+                // This only contains published items.
+                if (_autorouteEntries.TryGetContentItemId(alias, out var contentItemId))
                 {
-                    return Task.FromResult(contentItemId);
+                    return contentItemId;
+                } else // Not relevant to check against published here as we are only querying for the content item id for the path / alias provided.
```

Comment:
this could potentially cause lots of db lookups; as its going too query even if someone attempts to lookup an alias that doesn't exist ... whereas before; if it didnt exist this would just return null and let another IContentAliasProvider potentially pick it up :|

Reply:
I would not change this service.

Reply:
Reverted.
Noting that it wouldn't cause extra lookups as the `AutorouteAliasProvider` checks for `.StartsWith("slug")` before it performs any lookups.
So without this you can get an alias draft, but not an autoroute draft (it will have to be published to be in the `AutorouteEntries`)